### PR TITLE
prusa-slicer: 2.2.0 -> 2.3.0

### PR DESCRIPTION
**Pull request by Moredread**
###### Motivation for this change

Preparing for the coming 2.3.0 release, documenting/sharing what needs to change.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Moredread** [comment]
cc @mweinelt @ArdaXi @baracoder @thorstenweber83 as you were interested in the previous PRs

**Comment by vojta001** [comment]
Builds fine for me, but throws assertion errors from time to time and some UI parts are broken. It is still RC1 though

**Comment by anna328p** [comment]
v2.3.0 is out.

**Comment by Moredread** [comment]
I've updated the PR to 2.3.0, which uses GTK3. I hope I got all new dependencies...

**Comment by ArdaXi** [comment]
I was trying to get this working myself (I really wanted to try the monotonic infill) and these are much the same changes I made. I did also end up adding `hicolor-icon-theme` because it was complaining about it being missing in the logs, but I'm not sure if it being missing actually affects anything.

**Comment by Moredread** [comment]
> I was trying to get this working myself (I really wanted to try the monotonic infill) and these are much the same changes I made. I did also end up adding `hicolor-icon-theme` because it was complaining about it being missing in the logs, but I'm not sure if it being missing actually affects anything.

I have `hicolor-icon-theme` in my home-managers packages, so I didn't see that. I've added it to our inputs though as at least some packages do add it to their inputs. Not sure if this correct though...

**Comment by ArdaXi** [comment]
Isn't that the same argument as for `wrapGAppsHook`? That should also be unnecessary if the user's environment is set up properly, and it seems like NixOS has standardized on providing it. See discussion in #16285

**Comment by Moredread** [comment]
> Isn't that the same argument as for `wrapGAppsHook`? That should also be unnecessary if the user's environment is set up properly, and it seems like NixOS has standardized on providing it. See discussion in #16285

Thanks, I didn't know about that. :)

**Comment by vojta001** [comment]
`prusa-gcodeviewer` binary is not built, but prusa-slicer expects it to be

**Comment by Moredread** [comment]
> `prusa-gcodeviewer` binary is not built, but prusa-slicer expects it to be

Thanks! :) I've also added a application menu entry.

**Comment by cab404** [comment]
Result of `nixpkgs-review pr 107187` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>prusa-slicer</li>
    <li>super-slicer</li>
  </ul>
</details>

**Review by cab404 — APPROVED** [review]

**Comment by cab404** [comment]
Wow, their GUI looks nice in this update)
All functionality works, slicing and exporting doesn't crash like it does in SS 2.3 branches.

File: pkgs/applications/misc/prusa-slicer/default.nix [hunk 1]
```diff
@@ -1,37 +1,44 @@
...
 }:
 stdenv.mkDerivation rec {
   pname = "prusa-slicer";
-  version = "2.2.0";
+  version = "2.3.0";
 
   nativeBuildInputs = [
     cmake
+    copyDesktopItems
     pkgconfig
```
Comment (mweinelt) [inline pkgs/applications/misc/prusa-slicer/default.nix:12]:
```suggestion
    pkg-config
```

The current is an alias for the latter, and is deprecated.

File: pkgs/applications/misc/prusa-slicer/default.nix [hunk 2]
```diff
@@ -1,37 +1,44 @@
...
     openvdb
+    pcre
     systemd
     tbb
-    wxGTK31
+    wxGTK31-gtk3
     xorg.libX11
   ] ++ checkInputs;
 
+  doCheck = true;
```
Comment (mweinelt) [inline pkgs/applications/misc/prusa-slicer/default.nix:41]: :heart:

**Review by mweinelt — COMMENTED** [review]
That looks like a nice improvement. Unfortunately the printer is too far away in these days, so no real world testing. I clicked a bit around and found no immediate issue.

File: pkgs/applications/misc/prusa-slicer/default.nix (hunk 1 above)
Comment (Moredread) [reply pkgs/applications/misc/prusa-slicer/default.nix]: Done :)

**Comment by Moredread** [comment]
`prusa-slicer` segfaults when opening a gcode file with `File -> G-Code preview` after launching `prusa-gcodeviewer`. Does anyone else see this?

**Comment by mweinelt** [comment]
> `prusa-slicer` segfaults when opening a gcode file with `File -> G-Code preview` after launching `prusa-gcodeviewer`. Does anyone else see this?

Can't reproduce. Works for me.

**Comment by vojta001** [comment]
@Moredread It works for me. Can you, may be, share your GCODE with us?

**Comment by Moredread** [comment]
Thanks for checking, I guess in that case it's an upstream issue. I'll post  my gcode file later if I remember, but it shouldn't block this PR.

**Comment by Moredread** [comment]
It now works, no idea what has changed overnight. XD

**Review by mweinelt — APPROVED** [review]

**Comment by mweinelt** [comment]
Thanks @Moredread!

**Pull request merged**